sway/ipc-json.c

```diff
@@ -123,7 +144,19 @@ static void ipc_json_describe_output(struct sway_output *output,
 	}
 
 	json_object_object_add(object, "modes", modes_array);
-	json_object_object_add(object, "layout", json_object_new_string("output"));
+
+	struct sway_node *parent = node_get_parent(&output->node);
+	struct wlr_box parent_box = {0, 0, 0, 0};
+
+	if (parent != NULL) {
+		node_get_box(parent, &parent_box);
+	}
+
+	if (parent_box.width != 0 && parent_box.height != 0) {
+		double percent = ((double)output->width / (double)parent_box.width) 
```

Comment:
Only one of the operands needs to be casted to double. Also, there seems to a rogue trailing space here which you should definitely get rid of.

Also, does the percent property get added when the parent box is 0? If not, maybe it should have a default value.

Reply:
Oh wait, I missed this question. tbh I'm not sure either, because I don't really use it. Maybe we should just leave it for now and see if someone with a better use case for it has an idea.